### ISSUE TITLE
[java] LooseCoupling: not reported for a `var` local whose inferred type is a concrete collection

### ISSUE THREAD

**Issue by Chordrain** [pmd/pmd#6766]
**PMD Version:** 7.24.0
**Category:** [LooseCoupling](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#loosecoupling)
**Severity:** Warning (Best Practices)

### Summary

`LooseCoupling` flags local variables declared with a concrete collection implementation type (`HashSet`, `ArrayList`, …) instead of the corresponding interface. When the explicit type is replaced with `var`, the inferred static type is still exactly `HashSet<Integer>` (JLS §14.4.1), so the variable is just as coupled to the implementation type — yet the rule no longer reports it.

### Reproducer

`before/Sample.java` (flagged — correct):

```java
import java.util.HashSet;
public class Sample {
    public void m() {
        HashSet<Integer> set = new HashSet<Integer>();
        set.add(1);
        System.out.println(set.hashCode());
    }
}
```

`after/Sample.java` (not flagged — missed):

```java
import java.util.HashSet;
public class Sample {
    public void m() {
        var set = new HashSet<Integer>();
        set.add(1);
        System.out.println(set.hashCode());
    }
}
```

### Expected behavior

Both declarations have the same static type `HashSet<Integer>`, so the rule should treat them consistently — either report both or neither.

### Actual behavior

Explicit-type form → 1 violation:

```
before/Sample.java:4: LooseCoupling: Avoid using implementation types like 'HashSet'; use the interface instead
```

`var` form → 0 violations. The rule inspects the textual type node of the declaration and does not look through the type inferred for a `var` local.

**Comment by UncleOwen** [comment]
I'd argue that's expected behavior. The point of the rule is to make sure that you can replace `new HashSet<Integer>` with, say `new LinkedHashSet<Integer>` or `new TreeSet<Integer>`. You can still do that if you use `var`.

See the first sentence of the rule documentation:
> Excessive coupling to implementation types (e.g., `HashSet`) limits your ability to use alternate implementations in the future as requirements change.

**Comment by Chordrain** [comment]
Thanks for the explanation. I see your point that `var` still allows replacing `HashSet` with another implementation without changing the variable declaration.

I was thinking about the inferred type being the concrete implementation, but I agree that doesn't necessarily conflict with the rule's stated goal. Thanks for clarifying the intended behavior.
